File: examples/companion-packages/dotnet-csharp-nuget-app/src/AgentXM.Examples.PawMatch.CSharp/MatchPreferences.cs
using System.Collections.Frozen;

namespace AgentXM.Examples.PawMatch.CSharp;

internal sealed record MatchPreferences(
    bool HasKids,
    bool QuietHome,
    bool Active,
    bool FirstTime,
    bool MultiplePets,
    bool SmallHome)
{
    public IEnumerable<string> ActiveFlags()
    {
        if (HasKids) yield return "has-kids";
        if (QuietHome) yield return "quiet-home";
        if (Active) yield return "active";
        if (FirstTime) yield return "first-time";
        if (MultiplePets) yield return "multiple-pets";
        if (SmallHome) yield return "small-home";
    }

    public FrozenSet<string> ToFlagSet() =>
        ActiveFlags().ToFrozenSet(StringComparer.Ordinal);

    public bool IsEmpty =>
        !HasKids && !QuietHome && !Active && !FirstTime && !MultiplePets && !SmallHome;
}

File: examples/companion-packages/javascript-npm-app/src/match-preferences.js
// @ts-check

/**
 * @typedef {object} MatchPreferences
 * @property {boolean} hasKids
 * @property {boolean} quietHome
 * @property {boolean} active
 * @property {boolean} firstTime
 * @property {boolean} multiplePets
 * @property {boolean} smallHome
 */

/**
 * @param {MatchPreferences} prefs
 * @returns {Set<string>}
 */
export function activeFlagSet(prefs) {
  /** @type {Set<string>} */
  const set = new Set();
  if (prefs.hasKids) set.add("has-kids");
  if (prefs.quietHome) set.add("quiet-home");
  if (prefs.active) set.add("active");
  if (prefs.firstTime) set.add("first-time");
  if (prefs.multiplePets) set.add("multiple-pets");
  if (prefs.smallHome) set.add("small-home");
  return set;
}

/**
 * @param {MatchPreferences} prefs
 * @returns {boolean}
 */
export function isEmpty(prefs) {
  return (
    !prefs.hasKids &&
    !prefs.quietHome &&
    !prefs.active &&
    !prefs.firstTime &&
    !prefs.multiplePets &&
    !prefs.smallHome
  );
}
